crates/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -479,22 +479,82 @@ impl<'a> Resolver<'a> {
+        mut self, 
         let_stmt: crate::LetStatement
     ) -> (HirStatement, Vec<ResolverError>) {
-        match let_stmt.expression.kind {
-            ExpressionKind::Literal(_) => {
-                print!("*** 1\n");
-                let expression = self.resolve_expression(let_stmt.expression);
-                let definition = DefinitionKind::Global(expression);
-                (HirStatement::Let(HirLetStatement {
-                    pattern: self.resolve_pattern(let_stmt.pattern, definition),
-                    r#type: self.resolve_type(let_stmt.r#type),
-                    expression,
-                }), self.errors)
+        match let_stmt.expression.kind.clone() {
+            // Ensure it is a literal.
+            ExpressionKind::Literal(literal) => {
+                // If it is a literal array, check the elements are literals themselves.
+                match literal.clone() {
+                    // Check each element of a standard array.
+                    Literal::Array(ArrayLiteral::Standard(elems)) => {
+                        let mut type_error = false;
+                        for e in elems.iter() {
+                            match e.kind {
+                                ExpressionKind::Literal(_) => {}
+                                // This element is not a literal.
+                                _ => {
+                                    type_error = true;
+                                    self.push_err(ResolverError::Expected { 
+                                        span: let_stmt.expression.span, 
+                                        expected: "global array elements can only be array, bool, integer or field literals".to_owned(), 
+                                        got: e.kind.to_string(),
+                                    });
+                                }
+                            }
+                        }
+                        if type_error {
+                            return (HirStatement::Error, self.errors)
+                        }
+                        let expression = self.resolve_expression(let_stmt.expression);
+                        let definition = DefinitionKind::Global(expression);
+                        let hir_statement = HirStatement::Let(HirLetStatement {
+                            pattern: self.resolve_pattern(let_stmt.pattern, definition),
+                            r#type: self.resolve_type(let_stmt.r#type),
+                            expression,
+                        });
+                        (hir_statement, self.errors)
+                    }
+                    // Just check the type of the repeated element.
+                    Literal::Array(ArrayLiteral::Repeated { repeated_element, length: _}) => {
+                        match repeated_element.kind {
+                            ExpressionKind::Literal(_) => {
+                                let expression = self.resolve_expression(let_stmt.expression);
```

Comment:
You could possibly move this chunk into a separate func like `resolve_literal` as it is used above for a Standard ArrayLiteral and in the Non-array literal case below

Reply:
nice! this is a great clean up. done

Reply:
https://github.com/michaelneuder/noir/pull/2/commits/452a54d200e50f0f294b965b174726cef8e13bf3 has this diff. (ignore the nargo file, i removed it in the next commit)